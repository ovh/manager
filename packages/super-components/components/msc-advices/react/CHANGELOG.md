# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.1.1](https://github.com/ovh/manager/compare/@ovhcloud/msc-react-advices@1.1.0...@ovhcloud/msc-react-advices@1.1.1) (2023-12-05)

**Note:** Version bump only for package @ovhcloud/msc-react-advices





# [1.1.0](https://github.com/ovh/manager/compare/@ovhcloud/msc-react-advices@1.0.0...@ovhcloud/msc-react-advices@1.1.0) (2023-12-05)


### Features

* msc-advices ([670db53](https://github.com/ovh/manager/commit/670db53549b433db305fa4e6875d804dc2da56c3))
