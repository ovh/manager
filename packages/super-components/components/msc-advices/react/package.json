{
  "name": "@ovhcloud/msc-react-advices",
  "version": "1.0.0",
  "private": true,
  "description": "React specific wrapper for msc-advices super component",
  "keywords": [],
  "homepage": "https://github.com/ovh/manager/tree/master/packages/super-components/components/msc-advices#readme",
  "bugs": {
    "url": "https://github.com/ovh/manager/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ovh/manager.git",
    "directory": "packages/super-components/components/msc-advices/react"
  },
  "license": "Apache-2.0",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist/"
  ],
  "scripts": {
    "clean": "rimraf dist",
    "compile": "npm run compile:esm && npm run compile:cjs",
    "compile:cjs": "tsc -p tsconfig.cjs.json",
    "compile:esm": "tsc -p tsconfig.json",
    "msc:build": "npm run clean && npm run compile",
    "prepare": "yarn msc:build"
  },
  "dependencies": {
    "@ovhcloud/msc-stencil-advices": "*",
    "tslib": "*"
  },
  "devDependencies": {
    "@types/lodash": "4.14.197",
    "@types/react": "18.2.22",
    "@types/react-dom": "18.2.7",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "rimraf": "^3.0.2",
    "typescript": "5.1.3"
  },
  "peerDependencies": {
    "@ovhcloud/ods-components": "^16.0.1",
    "react": ">=18.0.0",
    "react-dom": ">=18.0.0"
  }
}
