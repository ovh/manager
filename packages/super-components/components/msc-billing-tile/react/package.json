{
  "name": "@ovhcloud/msc-react-billing-tile",
  "version": "1.1.1",
  "private": true,
  "description": "React specific wrapper for Billing Tile super component",
  "keywords": [],
  "homepage": "https://github.com/ovh/manager/tree/master/packages/super-components/components/msc-billing-tile#readme",
  "bugs": {
    "url": "https://github.com/ovh/manager/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ovh/manager.git",
    "directory": "packages/super-components/components/msc-billing-tile/react"
  },
  "license": "BSD-3-Clause",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist/"
  ],
  "scripts": {
    "build": "yarn msc:build",
    "clean": "rimraf dist",
    "compile": "yarn compile:esm && yarn compile:cjs",
    "compile:cjs": "tsc -p tsconfig.cjs.json",
    "compile:esm": "tsc -p tsconfig.json",
    "msc:build": "yarn clean && yarn compile",
    "prepare": "yarn msc:build"
  },
  "dependencies": {
    "@ovhcloud/msc-stencil-billing-tile": "^1.1.1",
    "tslib": "*"
  },
  "devDependencies": {
    "@types/lodash": "4.14.197",
    "@types/react": "18.2.22",
    "@types/react-dom": "18.2.7",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "rimraf": "^3.0.2",
    "typescript": "5.1.3"
  },
  "peerDependencies": {
    "react": ">=18.0.0",
    "react-dom": ">=18.0.0"
  }
}
