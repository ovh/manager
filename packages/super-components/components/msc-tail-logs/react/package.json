{
  "name": "@ovhcloud/msc-react-tail-logs",
  "version": "1.0.0",
  "private": true,
  "description": "React specific wrapper for Tail Logs super components",
  "keywords": [],
  "homepage": "https://github.com/ovh/manager/tree/master/packages/super-components/components/msc-tail-logs#readme",
  "bugs": {
    "url": "https://github.com/ovh/manager/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ovh/manager.git",
    "directory": "packages/super-components/components/msc-tail-logs/react"
  },
  "license": "Apache-2.0",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist/"
  ],
  "scripts": {
    "build": "yarn msc:build",
    "clean": "rimraf dist",
    "compile": "yarn compile:esm && yarn compile:cjs",
    "compile:cjs": "tsc -p tsconfig.cjs.json",
    "compile:esm": "tsc -p tsconfig.json",
    "msc:build": "yarn clean && yarn compile",
    "prepare": "yarn msc:build"
  },
  "dependencies": {
    "@ovhcloud/msc-stencil-tail-logs": "*",
    "tslib": "*"
  },
  "devDependencies": {
    "@types/lodash": "4.14.197",
    "@types/react": "18.2.22",
    "@types/react-dom": "18.2.7",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "rimraf": "^3.0.2",
    "typescript": "5.1.3"
  },
  "peerDependencies": {
    "react": ">=18.0.0",
    "react-dom": ">=18.0.0"
  }
}
