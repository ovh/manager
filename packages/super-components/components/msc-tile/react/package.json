{
  "name": "@ovhcloud/msc-react-tile",
  "version": "1.0.0",
  "private": true,
  "description": "React specific wrapper for Manager super components",
  "keywords": [],
  "license": "Apache-2.0",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist/"
  ],
  "scripts": {
    "build": "npm run clean && npm run compile",
    "clean": "rimraf dist",
    "compile": "npm run compile:esm && npm run compile:cjs",
    "compile:cjs": "tsc -p tsconfig.cjs.json",
    "compile:esm": "tsc -p tsconfig.json"
  },
  "dependencies": {
    "@ovhcloud/msc-stencil-tile": "^1.0.0",
    "tslib": "*"
  },
  "devDependencies": {
    "@types/react": "18.2.8",
    "@types/react-dom": "18.2.4",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "rimraf": "^3.0.2",
    "typescript": "5.1.3"
  },
  "peerDependencies": {
    "@ovhcloud/ods-stencil": "^13.0.1",
    "react": ">=18.0.0",
    "react-dom": ">=18.0.0"
  }
}
