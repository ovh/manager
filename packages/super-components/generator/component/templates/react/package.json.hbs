{
  "name": "@ovhcloud/msc-react-{{ packageName }}",
  "version": "1.0.0",
  "private": true,
  "description": "React specific wrapper for Manager super components",
  "keywords": [],
  "homepage": "https://github.com/ovh/manager/tree/master/packages/super-components/components/{{ componentTag }}#readme",
  "bugs": {
    "url": "https://github.com/ovh/manager/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ovh/manager.git",
    "directory": "packages/super-components/components/{{ componentTag }}/react"
  },
  "license": "Apache-2.0",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist/"
  ],
  "scripts": {
    "prepare": "yarn msc:build",
    "msc:build": "npm run clean && npm run compile",
    "clean": "rimraf dist",
    "compile": "npm run compile:esm && npm run compile:cjs",
    "compile:cjs": "tsc -p tsconfig.cjs.json",
    "compile:esm": "tsc -p tsconfig.json"
  },
  "dependencies": {
    "@ovhcloud/msc-stencil-{{ packageName }}": "*",
    "tslib": "*"
  },
  "devDependencies": {
    "@types/lodash": "4.14.197",
    "@types/react": "17.0.39",
    "@types/react-dom": "18.2.7",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "rimraf": "^3.0.2",
    "typescript": "5.1.3"
  },
  "peerDependencies": {
    "@ovhcloud/ods-stencil": "^14.1.0",
    "react": ">=18.0.0",
    "react-dom": ">=18.0.0"
  }
}
