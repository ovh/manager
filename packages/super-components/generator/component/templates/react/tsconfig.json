{
  "extends": "../../../tsconfig.react.json",
  "compilerOptions": {
    "outDir": "dist/esm",
    "declarationDir": "dist/types"
  },
  "include": ["src/**/*.ts", "src/**/*.tsx"]
}
