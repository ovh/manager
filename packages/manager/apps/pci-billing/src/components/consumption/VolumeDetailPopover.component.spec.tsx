import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import VolumeDetailPopover from './VolumeDetailPopover.component';

describe('VolumeDetailPopover', () => {
  const volumeDetail = {
    totalPrice: '1002',
    volumeId: 'volume-id',
    quantity: 2,
    region: 'region-1',
    type: 'type-1',
    amount: 123456,
    name: 'volume-name',
    status: 'enabled',
  };

  it('matches snapshot', () => {
    const { asFragment } = render(<VolumeDetailPopover row={volumeDetail} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
