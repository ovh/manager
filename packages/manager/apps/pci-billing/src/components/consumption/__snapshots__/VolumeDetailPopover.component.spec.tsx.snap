// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VolumeDetailPopover > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-3"
  >
    <osds-text
      class="block"
      color="text"
      level="body"
      size="400"
    >
      <strong>
        cpbc_volume_location : 
      </strong>
      region-1
    </osds-text>
    <osds-text
      class="block"
      color="text"
      level="body"
      size="400"
    >
      <strong>
        cpbc_volume_type : 
      </strong>
      type-1
    </osds-text>
    <osds-text
      class="block"
      color="text"
      level="body"
      size="400"
    >
      <strong>
        cpbc_volume_size : 
      </strong>
    </osds-text>
    <osds-text
      class="block"
      color="text"
      level="body"
      size="400"
    >
      <strong>
        cpbc_volume_state : 
      </strong>
      cloud_common_state_enabled
    </osds-text>
  </div>
</DocumentFragment>
`;
